[ANT-2034] Minimal system under test

JasonMarechal25 opened the pull request.

JasonMarechal25 committed:
InMemory modeler

JasonMarechal25 committed:
InMemory modeler test

JasonMarechal25 committed:
InMemory modeler test

JasonMarechal25 committed:
Fix build

JasonMarechal25 committed:
Fix test

JasonMarechal25 committed:
Format

JasonMarechal25 committed:
Format

JasonMarechal25 committed:
Fix

JasonMarechal25 committed:
Move fixture and use it in in memory modeler

JasonMarechal25 committed:
Move fixture and use it in in memory modeler

JasonMarechal25 committed:
Refacto

JasonMarechal25 committed:
Refacto

JasonMarechal25 committed:
Format

JasonMarechal25 committed:
Format

JasonMarechal25 committed:
Merge branch 'develop' into feature/4.3_inmemory

flomnes committed:
Unit test synthesis aggregation (EXP/MIN/MAX/STD) (#2828)

JasonMarechal25 committed:
Virtual destructor

JasonMarechal25 committed:
Minimal system unit test

JasonMarechal25 committed:
Minimal system unit test

JasonMarechal25 committed:
Discard changes to src/tests/resources/modeler/1_1/input/data-series/gen1_max_p_1_1.tsv

JasonMarechal25 committed:
Discard changes to src/tests/resources/modeler/1_1/input/data-series/gen1_prop_cost_1_1.tsv

JasonMarechal25 committed:
Discard changes to src/tests/resources/modeler/1_1/input/data-series/gen2_prop_cost_1_1.tsv

JasonMarechal25 committed:
Discard changes to src/tests/resources/modeler/1_1/input/data-series/load_1_1.tsv

JasonMarechal25 committed:
Discard changes to src/tests/resources/modeler/1_1/input/data-series/gen2_max_p_1_1.tsv

JasonMarechal25 committed:
Discard changes to src/tests/resources/modeler/1_1/input/data-series/load_1_1.tsv

JasonMarechal25 committed:
Discard changes to src/tests/resources/modeler/1_1/input/data-series/load_1_1.tsv

JasonMarechal25 committed:
Discard changes to src/tests/resources/modeler/1_1/input/data-series/load_1_1.tsv

JasonMarechal25 committed:
Discard changes to src/tests/resources/modeler/1_1/input/data-series/load_1_1.tsv

JasonMarechal25 committed:
Discard changes to src/tests/resources/modeler/1_1/input/model-libraries/lib_example_1_1.yml

JasonMarechal25 committed:
Discard changes to src/tests/resources/modeler/1_1/input/system.yml

JasonMarechal25 committed:
Fix

JasonMarechal25 committed:
Fix

JasonMarechal25 committed:
Fix

JasonMarechal25 committed:
Format

JasonMarechal25 committed:
Format

flomnes committed:
Force the export for criterion files in Expansion mode [ANT-3137] (#2821)

payetvin committed:
Add some unit tests for hydro (#2831)

payetvin committed:
Remove yuni limit on max threads (#2832)

JasonMarechal25 committed:
Fix missing ortools dll in windows zip asset (#2840)

payetvin committed:
Remove end to end tests on TS generator binary (#2843)

flomnes committed:
Document STS additional constraints @ v9.2.0 (#2847)

flomnes committed:
Remove unused members from spatial-aggregate.h (#2848)

JasonMarechal25 committed:
[ANT-3359] Scenario-group is optional in component and default to "default" (#2855)

flomnes committed:
Fix compilation warning (implicit double -> float conversion) (#2862)

flomnes committed:
Add changelog for v8.8.x (#2854)

flomnes committed:
Add changelog for v9.2.1 (#2853)

JasonMarechal25 committed:
[ANT-2034] Various warnings or linter fixes (#2861)

payetvin committed:
Simtest 9.3.0b (#2842)

guilpier-code committed:
Hydro monthly heuristic : add variable overflow to problem, onto v9.3.0-rc1 [ANT-3236] (#2837)

Attempt to fix ticket
[ANT-3236](https://gopro-tickets.rte-france.com/browse/ANT-3236) for
branch **v9.3.0-rc1**

**Note** :
This work was also done for branche **8.8.x** at[ this
location](https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/2819)

flomnes committed:
Add Gurobi to list of queried solvers [ANT-3381] (#2865)

payetvin committed:
Remove timeseries refresh [ANT-1335] (#2451)

Remove the refresh feature: no known users, add unnecessary complexity
to the code

TODO: 
- [x] Remove parallel test activation for every commit
- [x] Add doc

---------

Co-authored-by: Florian Omnès <florian.omnes@rte-france.com>

flomnes committed:
Add more doc for STS additional constraints (#2867)

flomnes committed:
Add STS additional constraints to infeasible problem analysis [ANT-3386] (#2869)

payetvin committed:
9.3 rc 2 (#2870)

payetvin committed:
Bump simtest to 9.3.0c (#2871)

guilpier-code committed:
or-tools : update to v9.13 [ANT-3348] (#2857)

Fixes [ticket
ANT-3348](https://gopro-tickets.rte-france.com/browse/ANT-3348)

---------

Co-authored-by: Florian OMNES <florian.omnes@rte-france.com>
Co-authored-by: Vincent Payet <vincent.payet@rte-france.com>

flomnes committed:
Add cucumber tests for v9.3.0 STS scenarization features [ANT-3037] (#2873)

flomnes reviewed (changes requested):
Very similar to #2856, is this a mistake ?

flomnes commented on src/tests/end-to-end/modeler/test-modeler.cpp:
```diff
@@ -58,7 +58,7 @@ class InMemoryLoader: public Antares::Solver::ILoader
     Antares::Solver::ModelerParameters loadParameters() override
     {
-        return {.solver = "sirius",
+        return {.solver = "xpress",
```
"sirius" (xpress pas dispo en CI)

payetvin committed:
Use correct keys for ccache, update ccache for windows (#2864)

flomnes committed:
Simplify applyToMatrix calls, remove useless CheckValidity template specializations (#2879)

payetvin committed:
Always save ccache (#2884)

flomnes committed:
Use std::vector<double> instead of double*, add const (#2878)

flomnes committed:
Cache chocolatey and installed utilities (Windows) (#2883)

Try to restore directory C:\ProgramData\chocolatey. Install choco &
utilities only if the cache restore fails.

flomnes committed:
Random TS numbers for STS inflows & additional constraints [ANT-3415] (#2882)

payetvin committed:
Save sts inflows in legacy GUI [ANT-3425] (#2886)

a-zakir committed:
[ANT-3409] Store short term time series number (#2881)

payetvin committed:
Better checking of TS sizes [ANT-3428] (#2888)

payetvin committed:
Bypass JIT only for STS [ANT-3425] (#2887)

sylvlecl committed:
doc(fmt): fix bullepoint formatting (#2890)


Some bulletpoint lists are currently not correctly formatted on RDT
because of missing whitelines.

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

flomnes committed:
Re-implement MPObjective::Value (temporary fix for Windows) (#2892)

flomnes committed:
Revert chocolatey cache (#2893)

Restoring the cache takes ~5min

Installing deps through chocolatey takes ~55s

a-zakir committed:
checking modeler variable bounds (#2844)

flomnes committed:
Remove year batches, launch jobs continually [ANT-2139] (#2718)

## Basic
- [X] Loop on years, instead of looping on batches
- [X] Implement `NumSpaceManager` mechanism, where a job requests an
available `numSpace` at the start and returns it at the end of its
execution
- [X] The code builds
- [X] No segfault when running with e.g `--force-parallel=8`

## Restore stuff
- [X] Restore year-by-year vartiables (mc-ind directory)
- [x] Restore synthesis variables for areas/links/clusters/etc. (mc-all
directory)
- [x] Restore synthesis variables for districts
- [X] Restore random generation
- [x] Restore `computeAnnualCostsStatistics` (should be easy)
- [x] Restore the progression display (for Antares Web & the Legacy GUI)
=> AntaresWeb parses the logs to estimate the progression
- [X] Restore Xpansion files single export #2726

## Result comparison
- [x] Compare (named) MPS files
- [X] Compare results mc-ind
- [x] Compare results mc-all
- [x] Check behavior with year filters (playlist)
- [ ] windows results mc-all

## Performance comparison
- [X] **Check performances for a real study**
- [x] Compare the memory peak on a real large study

## Pre-requisites
- [x] Remove in-memory TS generation and refresh

## Cleaning
- [ ]  Activate parallel tests in CI for extended tests

---------

Co-authored-by: Vincent Payet <vincent.payet@rte-france.com>
Co-authored-by: payetvin <113102157+payetvin@users.noreply.github.com>
Co-authored-by: Jason Maréchal <45510813+JasonMarechal25@users.noreply.github.com>

guilpier-code committed:
Infeasibility caused by hydro (attempt 2) [ANT-3122] (#2874)

Trial to fix [ticket
ANT-3122](https://gopro-tickets.rte-france.com/browse/ANT-3122)

A first attempt was made
[here](https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/2872).

Instead of changing the first PR, we created the new current PR because
:
- it felt easier to create a new PR
- we wanted to keep old changes contained in first PR for any purpose 

These (new) changes were imagined after failures occurring on the CI of
first PR (see failures
[here](https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/2872)).

**What was also done (regarding CI only)** : 
It's bad (we should have done that in separate PRs), but in this PR, we
also :
- moved archives (complete and solver only) creation and upload
(**.zip** on **Win** and **.tar.gz** on **Ubuntu**) to right after the
build step
- fix python scripts : about check integrity comparison, we switch
actual and reference results so that logs are correct in case we get
differences between the two.

---------

Co-authored-by: Vincent Payet <vincent.payet@rte-france.com>
Co-authored-by: Florian Omnès <florian.omnes@rte-france.com>

payetvin committed:
Run parallel tests on windows only in extended tests (#2894)

flomnes committed:
Use range-based for loops (#2897)

payetvin committed:
9.3-rc3 (#2895)

flomnes committed:
Add modeler features 8.1, 8.2 and 8.3 [ANT-3535] (#2902)

- Keep only TS1 for wind, thermal and load other TS are not used (3*40
Mb -> 3*1,5 Mb)
- For 8_1 & 8_2, add checks on values based on hand-made computations
- For 8_3 only check the objective value since time coupling introduces
multiple equivalent solutions.

payetvin committed:
Dynamic groups for thermal and renewable clusters [ANT-1672] (#2579)

This pull request includes several changes to the codebase, focusing on
refactoring, improving conditional logic, and updating versioning. The
most important changes include adding conditional checks to workflows,
removing redundant matrix test functions, replacing custom exceptions
with standard ones, and updating version numbers.

### Workflow Improvements:
*
[`.github/workflows/ubuntu.yml`](diffhunk://#diff-6640e8c668a6deeb426558faaa2b0f535d95814c09a05c044a1df27a008dd544R248):
Added conditional checks to run cucumber tests only if specific
environment variables are set and the job is not cancelled.
[[1]](diffhunk://#diff-6640e8c668a6deeb426558faaa2b0f535d95814c09a05c044a1df27a008dd544R248)
[[2]](diffhunk://#diff-6640e8c668a6deeb426558faaa2b0f535d95814c09a05c044a1df27a008dd544R288)
*
[`.github/workflows/windows-vcpkg.yml`](diffhunk://#diff-8e86046c750ae789851a44583d756b322d7908feaafb4776eb93ac4e331d1d13R249):
Added similar conditional checks for cucumber tests.
[[1]](diffhunk://#diff-8e86046c750ae789851a44583d756b322d7908feaafb4776eb93ac4e331d1d13R249)
[[2]](diffhunk://#diff-8e86046c750ae789851a44583d756b322d7908feaafb4776eb93ac4e331d1d13R289)

### Codebase Simplification:
*
[`src/libs/antares/array/include/antares/array/matrix.h`](diffhunk://#diff-ec412f98c794bbf244613bd015d928e0990628b9a30ef63762d044d69ac586baL544-L585):
Removed multiple matrix test functions that were redundant.
*
[`src/libs/antares/array/matrix.cpp`](diffhunk://#diff-f82f00ff935edcc1c83d0e7e8066fb4d7866840c256dcb5c287cc39ebd80c62eL2-L132):
Removed the corresponding implementations of the matrix test functions.

### Exception Handling:
*
[`src/libs/antares/include/antares/antares/Enum.hxx`](diffhunk://#diff-f4b7f6248f5568080139a2e1caed1b11e131e085113545919655ba34d22b3452R27-L30):
Replaced `AssertionError` with `std::runtime_error` for exception
handling.
[[1]](diffhunk://#diff-f4b7f6248f5568080139a2e1caed1b11e131e085113545919655ba34d22b3452R27-L30)
[[2]](diffhunk://#diff-f4b7f6248f5568080139a2e1caed1b11e131e085113545919655ba34d22b3452L41-R40)
[[3]](diffhunk://#diff-f4b7f6248f5568080139a2e1caed1b11e131e085113545919655ba34d22b3452L54-R53)
*
[`src/libs/antares/study/UnfeasibleProblemBehavior.cpp`](diffhunk://#diff-22ee2f6b03856b19380aded0fb1e8ab2351478ae5c5ba6907078eb50678ec92fL44-R43):
Updated exception handling to use `std::runtime_error` instead of
`AssertionError`.
[[1]](diffhunk://#diff-22ee2f6b03856b19380aded0fb1e8ab2351478ae5c5ba6907078eb50678ec92fL44-R43)
[[2]](diffhunk://#diff-22ee2f6b03856b19380aded0fb1e8ab2351478ae5c5ba6907078eb50678ec92fL62-R61)
[[3]](diffhunk://#diff-22ee2f6b03856b19380aded0fb1e8ab2351478ae5c5ba6907078eb50678ec92fL80-R79)
[[4]](diffhunk://#diff-22ee2f6b03856b19380aded0fb1e8ab2351478ae5c5ba6907078eb50678ec92fL100-R99)

### Version Updates:
*
[`simtest.json`](diffhunk://#diff-c2088552287e6bf08a1bf79787828d4e59d6a9b1010156e943cd7196e2365be6L2-R2):
Updated the version from "v9.2.0i" to "v9.2.0j".
*
[`src/CMakeLists.txt`](diffhunk://#diff-148715d6ea0c0ea0a346af3f6bd610d010d490eca35ac6a9b408748f7ca9e3f4L10-R10):
Updated the release candidate number from 9 to 10.

### Refactoring:
*
[`src/libs/antares/study/include/antares/study/parts/common/cluster.h`](diffhunk://#diff-6632e6418938cf399ed6b48a78bdcc573dd0538dafcd159f7881b864fb65980bL57-R63):
Changed `ClusterName` type from a custom type to `std::string`.
[[1]](diffhunk://#diff-6632e6418938cf399ed6b48a78bdcc573dd0538dafcd159f7881b864fb65980bL57-R63)
[[2]](diffhunk://#diff-6632e6418938cf399ed6b48a78bdcc573dd0538dafcd159f7881b864fb65980bL138-R140)
*
[`src/libs/antares/study/include/antares/study/parts/common/cluster_list.h`](diffhunk://#diff-8204a7789a02a029e28598b45a487326de846f7df1487f61d05d34af60538108L67-R69):
Updated methods to use `std::string` for cluster IDs.
[[1]](diffhunk://#diff-8204a7789a02a029e28598b45a487326de846f7df1487f61d05d34af60538108L67-R69)
[[2]](diffhunk://#diff-8204a7789a02a029e28598b45a487326de846f7df1487f61d05d34af60538108L86-R93)

These changes improve the maintainability and readability of the code
and ensure better conditional handling in workflows.

---------

Co-authored-by: GitHub Actions <actions@github.com>
Co-authored-by: Florian OMNES <florian.omnes@rte-france.com>
Co-authored-by: Florian OMNES <26088210+flomnes@users.noreply.github.com>
Co-authored-by: Jason Maréchal <45510813+JasonMarechal25@users.noreply.github.com>

JasonMarechal25 committed:
[ANT-2034] InMemory modeler test (#2827)

Refactoring to allow in memory modeler test.

- Move _modeler-lib_ outside _modeler_ exe scope
- _modeler-lib_ now expose a _Modeler_ and some interfaces
- _modeler_ exe depends on modeler-lib for _Modeler_ and _IFileWriter_
interface
- _load-modeler-files_ depends on _modeler-libs_ for _ILoader_ interface

---------

Co-authored-by: Florian Omnès <florian.omnes@rte-france.com>
Co-authored-by: Peter Mitri <peter.mitri@rte-france.com>

payetvin committed:
Bump simtest to 9.3.0f (#2908)

payetvin committed:
Add 2 future parameters in generaldata.ini [ANT-3504] (#2907)

flomnes committed:
Improve helper class `ThermalClusterConfig` (#2909)

- Remove default constructor (makes no sense)
- Use `std::shared_ptr` to guarantee no dangling references

flomnes committed:
Remove useless std::shared_ptr around `SimulationHandler` (#2911)

payetvin committed:
Documentation for 9.3 (#2913)

guilpier-code committed:
TS analyzer : removal (#2905)

This PR fixes [ticket
ANT-3567](https://gopro-tickets.rte-france.com/browse/ANT-3567)

---------

Co-authored-by: Florian Omnès <florian.omnes@rte-france.com>

payetvin committed:
Add sts constraint to migration guide (#2915)

payetvin committed:
Add sts inflow to scenario migration  (#2916)

flomnes committed:
[In-memory studies] Add & use helpers for short-term storage (#2910)

- Add helper for STS and STS additional constraints
- Add tests on scenarized additional constraints

JasonMarechal25 committed:
[ANT-2034] Minimal inmemory modeleur test (#2856)

Co-authored-by: Florian Omnès <florian.omnes@rte-france.com>

payetvin committed:
Improve doc for dynamic groups in migration guide (#2918)

sylvmara committed:
Ignore save functions when BUILD_UI=OFF [ANT-2115], remove study-updater, kirchoff-cbuilder (#2779)

Ref : ANT 2115

---------

Co-authored-by: Florian OMNES <florian.omnes@rte-france.com>

flomnes committed:
Test problems in-memory [ANT-3514] (#2917)

Add observer implementation allowing to tests problems in-memory.

This way, tests are more unitary since we only test the problem creation
from memory structures, not problem resolution, I/O and so forth.

![Untitled Diagram
drawio(4)](https://github.com/user-attachments/assets/0ef61a64-9418-407d-b952-4bbce1fee8ba)

payetvin committed:
Changelog 9.3.0 (#2920)

Co-authored-by: Florian Omnès <florian.omnes@rte-france.com>

a-zakir committed:
[ANT-3507] fix infeasible problem analyzer, remove "safe resolution mode" (#2898)

Co-authored-by: Florian Omnès <florian.omnes@rte-france.com>
Co-authored-by: payetvin <113102157+payetvin@users.noreply.github.com>
Co-authored-by: GitHub Actions <actions@github.com>
Co-authored-by: Florian OMNES <26088210+flomnes@users.noreply.github.com>
Co-authored-by: Jason Maréchal <45510813+JasonMarechal25@users.noreply.github.com>
Co-authored-by: Peter Mitri <peter.mitri@rte-france.com>

flomnes committed:
[DOC] Formatting (#2923)

JasonMarechal25 committed:
[ANT-2034] Load modeler scenario builder (#2863)

Co-authored-by: Peter Mitri <peter.mitri@rte-france.com>
Co-authored-by: Florian Omnès <florian.omnes@rte-france.com>
Co-authored-by: Abdoulbari Zaher <32519851+a-zakir@users.noreply.github.com>

JasonMarechal25 committed:
Fix rebase

JasonMarechal25 committed:
Fix rebase

JasonMarechal25 committed:
Format

JasonMarechal25 committed:
Discard changes to .github/workflows/centos7.yml

JasonMarechal25 committed:
Discard changes to .github/workflows/oracle8.yml

JasonMarechal25 committed:
Discard changes to .github/workflows/sonarcloud.yml

JasonMarechal25 committed:
Discard changes to .github/workflows/ubuntu.yml

JasonMarechal25 committed:
Discard changes to .github/workflows/windows-vcpkg.yml

JasonMarechal25 committed:
Merge branch 'develop' into feature/4.3_simple_system_test

JasonMarechal25 committed:
Discard changes to src/tests/resources/modeler/1_1/input/model-libraries/lib_example_1_1.yml

JasonMarechal25 committed:
Discard changes to src/tests/resources/modeler/1_1/input/data-series/gen2_prop_cost_1_1.tsv

JasonMarechal25 committed:
Discard changes to src/tests/resources/modeler/1_1/input/data-series/gen1_prop_cost_1_1.tsv

JasonMarechal25 committed:
Discard changes to src/tests/resources/modeler/1_1/input/data-series/load_1_1.tsv

JasonMarechal25 committed:
Discard changes to src/tests/resources/modeler/1_1/input/data-series/gen2_max_p_1_1.tsv

JasonMarechal25 committed:
Discard changes to src/tests/resources/modeler/1_1/input/data-series/gen1_max_p_1_1.tsv

JasonMarechal25 committed:
Fix

JasonMarechal25 committed:
Format

flomnes commented on src/tests/end-to-end/modeler/test-modeler.cpp:
```diff
@@ -116,23 +123,28 @@ class StubWriter: public Antares::Solver::IWriter
     InMemoryLoader<Test::Modeler::LinearProblemBuildingFixture> inMemoryLoader;
     StubWriter inMemoryWriter;
 
     const Antares::Solver::Modeler modeler(inMemoryLoader, inMemoryWriter);
     modeler.solve();
+    BOOST_CHECK_EQUAL(inMemoryWriter.solution_.objectiveValue, 0);
```
```suggestion
    const Antares::Solver::Modeler modeler(inMemoryLoader, inMemoryWriter);
    auto solution = modeler.solve();
    BOOST_CHECK_EQUAL(solution.objectiveValue, 0);
```

flomnes reviewed (commented):
The `solution` shouldn't be stored in the writer object, but rather returned by the `solve` method.

JasonMarechal25 committed:
Update src/tests/end-to-end/modeler/test-modeler.cpp

Co-authored-by: Florian Omnès <florian.omnes@rte-france.com>

JasonMarechal25 committed:
Discard changes to src/tests/end-to-end/modeler/test-modeler.cpp

JasonMarechal25 committed:
Revert "Discard changes to src/tests/end-to-end/modeler/test-modeler.cpp"

This reverts commit 0d11e38fa18e38ccf9886187d3d6ee696cbdd5b5.

JasonMarechal25 committed:
Revert "Update src/tests/end-to-end/modeler/test-modeler.cpp"

This reverts commit b74be53f851de8a7245c4fffb31191fba24f5405.
